src/app/api/server.ts

```diff
@@ -175,5 +225,27 @@ export async function initializeApi(agent: SaikiAgent, agentCardOverride?: Parti
         mcpTransport
     );
 
+    // Export current AgentConfig as YAML, omitting sensitive fields
+    app.get('/api/config.yaml', async (req, res) => {
+        try {
+            // Deep clone and sanitize
+            const rawConfig = agent.configManager.getConfig();
+            const exportConfig = JSON.parse(JSON.stringify(rawConfig));
+            // Remove sensitive API key
+            if (exportConfig.llm && 'apiKey' in exportConfig.llm) {
+                exportConfig.llm.apiKey = 'SET_YOUR_API_KEY_HERE';
+            }
```

Comment:
i don't think this will fetch the updated config after user changes prompts/model etc, because i don't think we are actually updating the base `agent.configManager.config` when those things change